pkg/apis/apps/validation/validation.go

```diff
@@ -108,6 +108,25 @@ func ValidateStatefulSetSpec(spec *apps.StatefulSetSpec, fldPath *field.Path, op
 					apps.OnDeleteStatefulSetStrategyType)))
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.StatefulSetAutoDeletePVC) {
+		requiredError := field.Required(fldPath.Child("persistentVolumeClaimDeletePolicy"), "")
+		if spec.PersistentVolumeClaimDeletePolicy == nil {
+			allErrs = append(allErrs, requiredError)
+		} else {
+			switch *spec.PersistentVolumeClaimDeletePolicy {
+			case "":
+				allErrs = append(allErrs, requiredError)
+			case apps.RetainPersistentVolumeClaimDeletePolicyType, apps.DeleteOnStatefulSetDeletionOnlyPersistentVolumeClaimDeletePolicyType, apps.DeleteOnScaledownOnlyPersistentVolumeClaimDeletePolicyType, apps.DeleteOnScaledownAndStatefulSetDeletionPersistentVolumeClaimDeletePolicyType:
```

Comment:
You could use `sets` to save some repetitiveness. For example: https://github.com/kubernetes/kubernetes/blob/a58dcb67df3b9ce86d5ed99e9a2a65acaacf3a60/pkg/apis/core/validation/validation.go#L1650

Reply:
Nice. Done.